### ISSUE TITLE
[NodeResourceTopology] Question: Who is responsible for updating the CR NodeResourceTopology

### ISSUE THREAD

**Issue by zhiyxu** [kubernetes-sigs/scheduler-plugins#452]
I used the NodeResourceTopologyMatch plugin for the first time and found it was not working as expected.
I have a node with 8GPU, each NUMA node has 4GPU. So I create CR as follows：

```
apiVersion: topology.node.k8s.io/v1alpha1
kind: NodeResourceTopology
metadata:
  name: node01
topologyPolicies: ["SingleNUMANodePodLevel"]
zones:
  - name: node-0
    type: Node
    resources:
      - name: nvidia.com/gpu
        capacity: 4
        allocatable: 4
        available: 4
  - name: node-1
    type: Node
    resources:
      - name: nvidia.com/gpu
        capacity: 4
        allocatable: 4
        available: 4
```

Then I create a deploy as follows, these 2 pods are allocated to different NUMA node:

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: k8s-deploy-test
spec:
  replicas: 2
  selector:
    matchLabels:
      name: k8s-deploy-test
  template:
    metadata:
      labels:
        name: k8s-deploy-test
      name: k8s-deploy-test
    spec:
      schedulerName: topo-aware-scheduler
      nodeSelector:
        kubernetes.io/hostname: node01
      containers:
      - name: app
        image: centos:7
        command: ["/bin/sleep"]
        args: ["3600"]
        resources:
          limits:
            nvidia.com/gpu: "3"
          requests:
            nvidia.com/gpu: "3"
```

There are currently 2GPU left in total, but distributed in different NUMA node. Then I create another deploy, which shouldn't be scheduled:

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: k8s-deploy-test-1
spec:
  replicas: 1
  selector:
    matchLabels:
      name: k8s-deploy-test-1
  template:
    metadata:
      labels:
        name: k8s-deploy-test-1
      name: k8s-deploy-test-1
    spec:
      schedulerName: topo-aware-scheduler
      nodeSelector:
        kubernetes.io/hostname: node01
      containers:
      - name: app
        image: centos:7
        command: ["/bin/sleep"]
        args: ["3600"]
        resources:
          limits:
            nvidia.com/gpu: "2"
          requests:
            nvidia.com/gpu: "2"
```

But the pod is scheduled successfully, and reject by kubelet:

```
# kubectl get pod | grep k8s-deploy
k8s-deploy-test-7b4b75454-g44nz               1/1     Running                 0          21h
k8s-deploy-test-7b4b75454-qz42s               1/1     Running                 0          21h
k8s-deploy-test-1-54f645bcc4-4rc2g            0/1     TopologyAffinityError   0          1s
k8s-deploy-test-1-54f645bcc4-7th2s            0/1     TopologyAffinityError   0          3s
k8s-deploy-test-1-54f645bcc4-wk4dx            0/1     TopologyAffinityError   0          2s
```

Is this a bug or is there something wrong with the way I use it？

**Comment by fromanirh** [comment]
Hi! A key prerequisite for `NodeResourceTopologyMatch` to work is to have always up to date `NodeResourceTopology` objects to make logic with. In the current code, the scheduler plugin depends on the freshness of the data. We have enhancements in progress (#437 and #439 ) to tolerate less frequent update.
The recommended approach to update the `NodeResourceTopology` objects is to deploy and configure `node-feature-discovery` in the cluster, notably the `nfd-topology-updater` daemonset available from version >= 0.10.
Alternatively, if NFD does not work, there is the [resource-topology-exporter daemonset](https://github.com/k8stopologyawareschedwg/resource-topology-exporter) which is being deprecated for NFD.

**Comment by fromanirh** [comment]
/close
Hi @zhiyxu I'm closing this issue because it's more like a question and I provided the answer. Feel free to reopen if there is anything to investigate
